src/time-slot-week-view/time-slot-week-view.tsx

```diff
@@ -0,0 +1,133 @@
+    /** Specifies should show or hide drop-down buttons and navigation arrows in header*/
+    showNavigationHeader?: boolean | undefined;
+    /** Specifies should allow selecting date on weekly-calendar component */
+    enableSelection?: boolean | undefined;
+    /** Specifies which date should be included in the rendered calendar component (date will not be selected) formate: YYYY-DD-MM*/
+    currentCalendarDate?: string | undefined;
+    /** Dates to disable in `YYYY-MM-DD` format. Example: `["2023-04-30"]` */
+    disabledDates?: string[] | undefined;
+    /** Specifies any callback to execute when user click on rendered slots on weekly-calendar component */
+    onSlotClick?: (date: string, timeSlot: TimeSlot) => void | undefined;
+    onChange?: (value: string) => void | undefined;
+    onWeekDisplayChange?: ((value: YearMonthWeekDisplay) => void) | undefined;
+}
+
+interface YearMonthWeekDisplay {
+    week: { firstDayOfWeek: string; lastDayOfWeek: string };
+    year: number;
+    month: number;
+}
+
+export const TimeSlotWeekView = ({
+    disabledDates,
+    onWeekDisplayChange,
+    onChange,
+    value,
+    minDate,
+    maxDate,
+    slots,
+    showNavigationHeader = true,
+    enableSelection,
+    onSlotClick,
+    currentCalendarDate,
+    ...otherProps
+}: TimeSlotWeekViewProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const [selectedDate, setSelectedDate] = useState<string>(); // YYYY-MM-DD
+    const dateFormat = "YYYY-MM-DD";
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+
+    const handleDateSelect = (value: Dayjs) => {
+        const stringValue = value.format(dateFormat);
+        setSelectedDate(stringValue);
+        onChange(stringValue);
+    };
+
+    const handleOnSlotClick = (date: string, slot: TimeSlot) => {
+        if (onSlotClick) {
+            onSlotClick(date, slot);
+        }
+    };
+
+    const performOnCalendarDate = (value: Dayjs) => {
+        if (onWeekDisplayChange) {
+            const returnValue = {
+                week: {
+                    firstDayOfWeek: value.startOf("week").format(dateFormat),
+                    lastDayOfWeek: value.endOf("week").format(dateFormat),
+                },
+                month: value.month() + 1,
+                year: value.year(),
+            };
+
+            onWeekDisplayChange(returnValue);
+        }
+    };
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+
+    const renderContent = () => {
+        return (
+            <Wrapper>
+                <CalendarManager
```

Comment:
can I confirm which week is supposed to display when a month/year is selected?

Reply:
It should be based on date, since only month/year changes in the calendar date. The week that includes the day of the current calendar date should be dispalyed.